server/build_event_protocol/build_event_handler/build_event_handler.go

```diff
@@ -159,14 +159,17 @@ func (e *EventChannel) writeCompletedBlob(ctx context.Context, blobID string, in
 }
 
 func (e *EventChannel) MarkInvocationDisconnected(ctx context.Context, iid string) error {
+	if !e.beValues.BuildFinished() {
+		return e.FinalizeInvocation(iid)
+	}
 	e.statusReporter.ReportDisconnect(ctx)
 
 	if err := e.pw.Flush(ctx); err != nil {
 		return err
 	}
 	invocation := &inpb.Invocation{
 		InvocationId:     iid,
-		InvocationStatus: inpb.Invocation_DISCONNECTED_INVOCATION_STATUS,
```

Comment:
I think this should flip back to DISCONNECTED

Reply:
Oh whoa oops, thanks for the catch.